base/bucket_gocb.go

```diff
@@ -1334,13 +1333,31 @@ func (bucket *CouchbaseBucketGoCB) GetWithXattr(k string, xattrKey string, rv in
 				return false, gocb.ErrKeyNotFound, uint64(0)
 			}
 			cas = uint64(res.Cas())
-			return false, nil, cas
 
 		default:
 			shouldRetry = isRecoverableReadError(lookupErr)
 			return shouldRetry, lookupErr, uint64(0)
 		}
 
+		if userXattrKey != "" {
+			userXattrCas, err := bucket.GetXattr(k, userXattrKey, uxv)
+			switch pkgerrors.Cause(err) {
+			case gocb.ErrKeyNotFound, gocb.ErrSubDocBadMulti:
```

Comment:
Would ErrKeyNotFound represent someone deleting the document in the time since the LookupIn was called?  Do we want to have GetWithXattr return that error in that case?

Reply:
I think you're right. Need to have a review of this. I think I wanted 'No xattr key' but the subdocbadmulti takes care of that.

Reply:
I don't think we actually want to return an error in this case. I think we want to return retry 'true'. As we'll want to still grab the tombstone xattr info. Like the `ErrSubDocMultiPathFailureDeleted` case

Reply:
GetXattr is setting the AccessDeleted subdoc flag, isn't it?  Would we actually get an ErrKeyNotFound if there's a server tombstone?

Reply:
Yeah, you're correct.